### DISCUSSION TITLE
Web worker compatibility

### DISCUSSION THREAD

**Discussion by Maxzurek** [kylebarron/parquet-wasm#270] [Q&A]
Hi everyone.

First, I wanted to thank the author for the library, it is great. 
I was wondering if this library is compatible with web workers? I am using React with vite.js, have configured the vite-plugin-wasm with ES6 but I can't get my web worker to work when importing from parquet-wasm/bundler/arrow2. Outside of the worker, everything works perfectly (except for large parquet files, it looks like im running out of memory or something).

Tank you for your answers.

**Comment by kylebarron** [comment]
There's no reason why I'd expect it not to work in a web worker, but I've never tried myself. If you find out any details let me know.

With large parquet files you do run out of memory because there's an extra copy to get the data out of wasm memory. That should be fixable in the future with [arrow-js-ffi](https://github.com/kylebarron/arrow-js-ffi), once it gets more stable

**Reply by Maxzurek** [reply]
Thank you for your quick reply!

The issue was on my side. I was instantiating the worker object and passing it to a function that returned a promise in which the worker was started, By instantiating the worker inside the executor, I fixed my issue and everything is now working perfectly!

I'll keep an eye on [arrow-js-ffi](https://github.com/kylebarron/arrow-js-ffi).

Again, thank you!
